sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/statsEstimation/UnionEstimationSuite.scala

```diff
@@ -0,0 +1,240 @@
+        attrDate -> ColumnStat(min = Some(1), max = Some(4)),
+        attrTimestamp -> ColumnStat(min = Some(1L), max = Some(4L))))
+
+    val s3 = 2.toShort
+    val s4 = 6.toShort
+    val b3 = 2.toByte
+    val b4 = 6.toByte
+    val columnInfo1: AttributeMap[ColumnStat] = AttributeMap(
+      Seq(
+        AttributeReference("cint1", IntegerType)() -> ColumnStat(
+          distinctCount = Some(2),
+          min = Some(3),
+          max = Some(6),
+          nullCount = Some(0),
+          avgLen = Some(8),
+          maxLen = Some(8)),
+        AttributeReference("cdouble1", DoubleType)() -> ColumnStat(
+          distinctCount = Some(2),
+          min = Some(2.0),
+          max = Some(7.0),
+          nullCount = Some(0),
+          avgLen = Some(8),
+          maxLen = Some(8)),
+        AttributeReference("cshort1", ShortType)() -> ColumnStat(min = Some(s3), max = Some(s4)),
+        AttributeReference("clong1", LongType)() -> ColumnStat(min = Some(2L), max = Some(6L)),
+        AttributeReference("cbyte1", ByteType)() -> ColumnStat(min = Some(b3), max = Some(b4)),
+        AttributeReference("cfloat1", FloatType)() -> ColumnStat(
+          min = Some(2.2f),
+          max = Some(6.1f)),
+        AttributeReference("cdecimal1", DecimalType(5, 4))() -> ColumnStat(
+          min = Some(Decimal(14.5)),
+          max = Some(Decimal(19.9))),
+        AttributeReference("cdate1", DateType)() -> ColumnStat(min = Some(3), max = Some(6)),
+        AttributeReference("ctimestamp1", TimestampType)() -> ColumnStat(
+          min = Some(3L),
+          max = Some(6L))))
+
+    val child1 = StatsTestPlan(
+      outputList = columnInfo.keys.toSeq.sortWith(_.exprId.id < _.exprId.id),
+      rowCount = 2,
+      attributeStats = columnInfo,
+      size = sz)
+
+    val child2 = StatsTestPlan(
+      outputList = columnInfo1.keys.toSeq.sortWith(_.exprId.id < _.exprId.id),
+      rowCount = 2,
+      attributeStats = columnInfo1,
+      size = sz)
+
+    val union = Union(Seq(child1, child2))
+
+    val expectedStats = logical.Statistics(
+      sizeInBytes = 2 * 1024,
+      rowCount = Some(4),
+      attributeStats = AttributeMap(
+        Seq(
+          attrInt -> ColumnStat(min = Some(1), max = Some(6)),
+          attrDouble -> ColumnStat(min = Some(2.0), max = Some(7.0)),
+          attrShort -> ColumnStat(min = Some(s1), max = Some(s4)),
+          attrLong -> ColumnStat(min = Some(1L), max = Some(6L)),
+          attrByte -> ColumnStat(min = Some(b1), max = Some(b4)),
+          attrFloat -> ColumnStat(min = Some(1.1f), max = Some(6.1f)),
+          attrDecimal -> ColumnStat(min = Some(Decimal(13.5)), max = Some(Decimal(19.9))),
+          attrDate -> ColumnStat(min = Some(1), max = Some(6)),
+          attrTimestamp -> ColumnStat(min = Some(1L), max = Some(6L)))))
+    assert(union.stats === expectedStats)
+  }
+
+  test("col stats estimation for decimal data type with different precision and scale") {
```

Comment:
Ah, my bad... We can safely assume input decimals have the same precision&scale in analyzed plans, so we don't need this test case.

Reply:
Removed